### ISSUE TITLE
Upgrade V1.7.50 uses 30% more memory than before

### ISSUE THREAD

**Issue by foreverm10** [VictoriaMetrics/VictoriaMetrics#2382]
**The basic information**
 

This vmstorage instance, write QPS is 740K, after upgrade, Memory usage increased from 14GB to 18GB, cache usage increased from 6GB to 9GB, heap usage increased from approximately 5GB, indexDB /dataBlocks cache number increased from 20K to 70K,index/dataBlocks cache hit ratio decreased from 99.3% to 97%. Slow insertion rate is twice as large as before: about from 40 k to 80 k, observation pprof to analyze heap memory that 70% of the heap memory is used to deal with bytesutil. ResizeNoCopyNoOVerallocate function

**Comment by foreverm10** [comment]
The original version was V1.63.0

**Comment by hagen1778** [comment]
Hi @foreverm10! 
How much time has passed since the update when you took these observations? What is the cache hit rate for `storage/tsid` before and after update?

**Comment by foreverm10** [comment]
six days has passed since the update ,   the cache hit rate for storage/tsid before is   99.8% and after is 99.6%

**Comment by foreverm10** [comment]
> Hi @foreverm10! How much time has passed since the update when you took these observations? What is the cache hit rate for `storage/tsid` before and after update?

six days has passed since the update , the cache hit rate for storage/tsid before is 99.8% and after is 99.6%

**Comment by hagen1778** [comment]
What was the prev version you upgrade from? 

> cache usage increased from 6GB to 9GB

Is this the total mem usage for all caches? Could you please check the tsid cache size before and after upgrade? Could you please collect and attach memory profile to this ticket or send it to roman @ victoriametrics.com?

**Comment by foreverm10** [comment]
yes ,it's total mem usage for all caches, tsid cache before is 4.9GB,  after is 4.6GB , It's not a big change

**Comment by foreverm10** [comment]
The original version was V1.63.0

**Comment by hagen1778** [comment]
>  tsid cache before is 4.9GB, after is 4.6GB

This could be a reason of `storage/tsid before is 99.8% and after is 99.6%`.

In v1.73 cache sizes has changed in order to address issues with performance. However, there are flags which allow to set the cache size manually:

- allow overriding default limits for the following in-memory caches, which usually occupy the most memory:
storage/tsid - the cache speeds up lookups of internal metric ids by metric_name{labels...} during data ingestion. The size for this cache can be tuned with -storage.cacheSizeStorageTSID command-line flag.
indexdb/dataBlocks - the cache speeds up data lookups in <-storageDataPath>/indexdb files. The size for this cache can be tuned with -storage.cacheSizeIndexDBDataBlocks command-line flag.
indexdb/indexBlocks - the cache speeds up index lookups in <-storageDataPath>/indexdb files. The size for this cache can be tuned with -storage.cacheSizeIndexDBIndexBlocks command-line flag. See also [cache tuning docs](https://docs.victoriametrics.com/#cache-tuning). See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/1940).

See https://docs.victoriametrics.com/CHANGELOG.html#v1731

Could you compare which caches started to use more memory than before?

**Comment by foreverm10** [comment]
I'm changing parameters - storage. CacheSizeStorageTSID, but doesn't work

**Comment by hagen1778** [comment]
> I'm increasing parameters - storage. CacheSizeStorageTSID, but doesn't work ，and the hit ratio decreased

That's strange. Could you please verify that startup logs contain message similar to the following:
```
 info    VictoriaMetrics/lib/logger/flag.go:20   flag "storage.cacheSizeStorageTSID"=<your_value>
```
Please also note, that change to the cache size will result into cache rebuild. The rebuild happens in the first minutes of the storage uptime. You can see how it slowly rebuilds/fills up on the Grafana dashboard.
